packages/docusaurus-theme-common/src/utils/blogUtils.ts

```diff
@@ -30,3 +30,22 @@ export function useVisibleBlogSidebarItems(
     [items, pathname],
   );
 }
+
+export function formatBlogPostDate(
+  locale: string,
+  date: Date | string,
+  calendar: string,
+): string {
+  try {
+    return new Intl.DateTimeFormat(locale, {
+      day: 'numeric',
+      month: 'long',
+      year: 'numeric',
+      timeZone: 'UTC',
+      calendar,
+    }).format(typeof date === 'string' ? new Date(date) : date);
+  } catch (err) {
+    console.error('Failed to format date', err);
+    throw err;
+  }
```

Comment:
```suggestion
  } catch (err) {
    throw new Error(`Failed to format date ${date}`,{cause: err as Error});
  }
```

Whenever possible, use error cause.
The existing code didn't use it because error cause was not well-supported before

I'm not sure we need a try/catch anymore. At the time it was added, it was probably because Intl wasn't widely supported in Node.js/Browsers